src/bridge/store/txnConfirmations.js

```diff
@@ -0,0 +1,39 @@
+import React, { forwardRef, useContext, useState } from 'react';
+
+export const TxnConfirmationsContext = React.createContext(null);
+
+export function TxnConfirmationsProvider({ children }) {
+  const [confirmations, setConfirmations] = useState({});
+
+  const setTxnConfirmations = (txnHash, txnConfirmations) => {
+    setConfirmations({
+      ...confirmations,
+      [txnHash]: txnConfirmations,
+    });
+  };
+
+  return (
+    <TxnConfirmationsContext.Provider
+      value={{
+        txnConfirmations: confirmations,
+        setTxnConfirmations,
+      }}>
+      {children}
+    </TxnConfirmationsContext.Provider>
+  );
+}
+
+// HOC version
+export const withTxnConfirmations = Component =>
+  forwardRef((props, ref) => (
+    <TxnConfirmationsContext.Consumer>
+      {txnConfirmations => (
+        <Component ref={ref} {...txnConfirmations} {...props} />
+      )}
+    </TxnConfirmationsContext.Consumer>
+  ));
+
+// Hook version
+export function useTxnConfirmations() {
+  return useContext(TxnConfirmationsContext);
+}
```

Comment:
"HOC version" is "do this as/with a component", whereas "hook version" is "do this as hook function", right?

The hook version here doesn't get used anywhere yet, but we'll be working towards that?

Reply:
@Fang- correct. a HOC wraps a component and provides some extra props. the HOC method is the most compatible with the existing code, but hooks are a replacement for the same behavior, so we should move towards that if we end up happy with them.